backend/homework/4/questions/src/main/java/question2/TicketReservation.java

```diff
@@ -0,0 +1,57 @@
+package question2;
+
+import com.ganesh.LogMaster;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.List;
+
+public class TicketReservation {
+
+    private static final int CONFIRMEDLIST_LIMIT = 10;
+    private static final int WAITINGLIST_LIMIT = 3;
+
+    private List<Passenger> confirmedList = new ArrayList<>();
+    private Deque<Passenger> waitingList = new ArrayDeque<>();
+
+    public List<Passenger> getConfirmedList() {
+        return confirmedList;
+    }
+    public boolean bookFlight(String firstName, String lastName, int age, String gender, String travelClass, String confirmationNumber) {
+        if(confirmedList.size() == CONFIRMEDLIST_LIMIT && waitingList.size() == WAITINGLIST_LIMIT){
+            return false;
+        }
```

Comment:
Handle validations properly. Lets try to make it more informative for debugging.